RecoLocalMuon/GEMSegment/plugins/ME0SegAlgoRU.cc

```diff
@@ -0,0 +1,464 @@
+/**
+ * \file ME0SegAlgRU.cc
+ *
+ *  \author M. Maggi for ME0
+ *  \from V.Palichik & N.Voytishin
+ *  \some functions and structure taken from SK algo by M.Sani and SegFit class by T.Cox
+ */
+#include "ME0SegAlgoRU.h"
+#include "MuonSegFit.h"
+#include "Geometry/GEMGeometry/interface/ME0Layer.h"
+#include "DataFormats/GeometryVector/interface/GlobalPoint.h"
+#include "DataFormats/Math/interface/deltaPhi.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include <algorithm>
+#include <cmath>
+#include <iostream>
+#include <string>
+
+#include <Math/Functions.h>
+#include <Math/SVector.h>
+#include <Math/SMatrix.h>
+
+
+
+ME0SegAlgoRU::ME0SegAlgoRU(const edm::ParameterSet& ps)
+  : ME0SegmentAlgorithmBase(ps), myName("ME0SegAlgoRU") {
+
+	allowWideSegments = ps.getParameter<bool>("allowWideSegments");
+	doCollisions = ps.getParameter<bool>("doCollisions");
+	stdParameters.maxChi2Additional = ps.getParameter<double>("maxChi2Additional");
+	stdParameters.maxChi2Prune      = ps.getParameter<double>("maxChi2Prune"     );
+	stdParameters.maxChi2GoodSeg    = ps.getParameter<double>("maxChi2GoodSeg"   ) ;
+	stdParameters.maxPhiSeeds       = ps.getParameter<double>("maxPhiSeeds"      ) ;
+	stdParameters.maxPhiAdditional  = ps.getParameter<double>("maxPhiAdditional" );
+	stdParameters.maxETASeeds       = ps.getParameter<double>("maxETASeeds"      );
+	stdParameters.maxTOFDiff        = ps.getParameter<double>("maxTOFDiff"      );
+	stdParameters.minNumberOfHits        = ps.getParameter<unsigned int>("minNumberOfHits"      );
+	stdParameters.requireBeamConstr = true;
+
+	wideParameters = stdParameters;
+	wideParameters.maxChi2Prune     *=2;
+	wideParameters.maxChi2GoodSeg   *=2;
+	wideParameters.maxPhiSeeds      *=2;
+	wideParameters.maxPhiAdditional *=2;
+	wideParameters.minNumberOfHits  +=1;
+
+
+	displacedParameters = stdParameters;

```

Comment:
In the CSC algorithm they were scaled wrt the config parameters (as is done here). I think it makes sense to do it separately. I would probably wait for making them configuration parameters until we actually use these extra iterations. At this point they are here for future development but disabled as we have not had time to optimize them/determine if they are necessary.